spec/page_spec.rb

```diff
@@ -107,6 +107,13 @@ def foo?
         page.load { |p| p.foo? && true }
       end
 
+      it 'should allow to load html and yields itself' do
+        class Page < SitePrism::Page; end
+        page = Page.new
+        expect(page).to receive(:foo?)
+        expect { page.load('<html/>') { |p| p.foo? && true } }.to_not raise_error
```

Comment:
I'd prefer you just passing in the html - remove the block argument as that's not under test here.

Also could we verify the yielded value is the string argument itself.

Reply:
@luke-hill maybe I mis-understood you.  Based on the test name I think testing that the block is executed is what s/he intended.  
That being said I don't think that this test actually proves what it intends to.  If the block is completely ignored the test will still pass.

I've never used this feature, myself, but based on the original issue it seems it could be something like:
`expect { page.load('<html>hi</html>') { |p| p.text } }.to eq 'hi'`

Reply:
Good point. I missed that. But yeh we definitely need a yielded arg